storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -95,13 +108,86 @@ func (s *commitLogSource) Read(
 	defer iter.Close()
 
 	var (
-		unmerged    = make(map[uint32]map[ts.Hash]encoderMap)
+		numShards = len(shardsTimeRanges)
+		unmerged  = make([]map[ts.Hash]encoderMap, numShards, numShards)
```

Comment:
Oh I see the issue now. I guess I didn't run into this issue because I was always using 8192 shards in my runs.

Instead of using a map, can I just initialize unmerged to MAX_SHARDS+1? I'm accessing by index, not appending, so then it should do the right thing?